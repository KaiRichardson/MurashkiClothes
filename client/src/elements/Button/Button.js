import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

const Button = ({ children, dark, ...rest }) => (
  <button className={`btn${dark ? '--dark' : ''}`} {...rest}>
    {children}
  </button>
);

export default Button;

Button.propTypes = {
  children: PropTypes.any.isRequired,
  dark: PropTypes.bool,
  rest: PropTypes.any
};
