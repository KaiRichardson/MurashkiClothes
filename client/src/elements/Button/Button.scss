@import '../../assets/sass/main.scss';

.btn {
  --bg: var(--clr-white);
  --txtColor: var(--clr-black);

  padding: $spacing(sm) $spacing(md);

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);

  &:hover {
    --bg: var(--clr-black);
    --txtColor: var(--clr-white);
  }

  /* Dark Button */
  &--dark {
    --bg: var(--clr-black);
    --txtColor: var(--clr-white);

    &:hover {
      --bg: var(--clr-grey);
    }
  }
}
