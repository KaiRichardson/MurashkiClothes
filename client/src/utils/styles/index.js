import spacing from './spacing';
import transition from './transition';

export * from './absolute';
export * from './colors';
export * from './fade';

export { spacing, transition };
