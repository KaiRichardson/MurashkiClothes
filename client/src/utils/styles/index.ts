import elevation from './elevation';
import spacing from './spacing';
import transition from './transition';

export * from './absolute';
export * from './colors';
export * from './fade';

export { elevation, spacing, transition };
