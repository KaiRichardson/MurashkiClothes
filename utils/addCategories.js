const Products = require('../models/Products');

module.exports = addCategories = async() => {

}