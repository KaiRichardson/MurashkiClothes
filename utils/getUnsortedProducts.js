const Products = require('../models/Products');

module.exports = getUnsortedProducts = async(res, printProducts) => {

  try {
    const products = await Products.find({}, '-category -name -imgUrls');

    const prodIDs = [];
    
    products.forEach(e => prodIDs.push(e._extID));

    const prodData = printProducts.filter(p => {
      if (prodIDs.indexOf(p.id) === -1) return true;

      return false;
    })

    //Check 0th element for null/falsey value since map will just return an array with null if there are no matches
    if(!prodData[0]) {
      return res.status(204).send('No products to add at this time!');
    }

    return res.status(200).json(prodData);

  } catch (err) {
    console.log(err);
    return res.status(500).send(`Error getting products: ${err}`)
  }
    
    

}