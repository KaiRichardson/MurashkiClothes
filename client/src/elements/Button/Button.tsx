import React from 'react';
import styled, { css } from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { spacing, transition, white, black, grey, lightGrey, red } from 'utils';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

export const Button = styled.button<{ dark?: boolean; trans?: boolean; brand?: boolean }>`
  --bg: ${white};
  --txtColor: ${black};

  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};
  border: none;
  ${transition({})};

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}

    /* Brand Color Button */
    ${props =>
      props.brand &&
      css`
        --bg: ${black};
        --txtColor: ${white};

        &:hover {
          --bg: ${red};
        }
      `}

    /* Transparent Button */
    ${props =>
      props.trans &&
      css`
        --txtColor: ${lightGrey};
        --bg: transparent;

        &:hover {
          --txtColor: ${black};
          --bg: transparent;
        }
      `}
`;

export const ButtonLink = styled(({ dark, trans, brand, ...rest }) => <Link {...rest} />)<{
  dark?: boolean;
  trans?: boolean;
  brand?: boolean;
}>`
  /* Default Button */
  --bg: ${white};
  --txtColor: ${black};

  background: var(--bg);
  color: var(--txtColor);

  text-align: center;
  display: inline-block;
  border: none;
  text-decoration: none;
  ${transition({})};
  cursor: pointer;
  padding: ${spacing.sm} ${spacing.md};
  

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}

    /* Brand Color Button */
    ${props =>
      props.brand &&
      css`
        --bg: ${black};
        --txtColor: ${white};

        &:hover {
          --bg: ${red};
        }
      `}

  /* Transparent Button */
  ${props =>
    props.trans &&
    css`
      --txtColor: ${lightGrey};
      --bg: transparent;

      &:hover {
        --txtColor: ${black};
        --bg: transparent;
      }
    `}
`;

Button.propTypes = {
  dark: PropTypes.bool,
  trans: PropTypes.bool
};

ButtonLink.propTypes = {
  dark: PropTypes.bool,
  trans: PropTypes.bool
};
