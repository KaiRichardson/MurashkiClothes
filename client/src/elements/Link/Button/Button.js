import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

import { spacing, transition, white, black, grey } from './node_modules/utils';

/*
    @desc Button component.
        Add optional dark prop for dark button, default is light.
*/

export const Button = styled.button`
  --bg: ${white};
  --txtColor: ${black};

  padding: ${spacing.sm} ${spacing.md};

  /* Default Button */
  background: var(--bg);
  color: var(--txtColor);
  border: none;
  ${transition({})};

  &:hover {
    --bg: ${black};
    --txtColor: ${white};
  }

  /* Dark Button */
  ${props =>
    props.dark &&
    css`
      --bg: ${black};
      --txtColor: ${white};

      &:hover {
        --bg: ${grey};
      }
    `}
`;

Button.propTypes = {
  dark: PropTypes.bool
};
