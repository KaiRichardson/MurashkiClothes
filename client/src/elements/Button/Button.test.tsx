import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import 'jest-styled-components';

import { Button, ButtonLink } from './Button';
import { red } from 'utils';

describe('<Button /> tests', () => {
  it('renders a default button', () => {
    //* Act
    const { getByTestId } = render(<Button data-testid='testButton'>Test Button</Button>);

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#000');
    expect(testButton).toHaveStyleRule('color', 'var(--txtColor)');

    expect(testButton).toHaveStyleRule('--bg', '#fff');
    expect(testButton).toHaveStyleRule('background', 'var(--bg)');

    // Hover tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff', {
      modifier: ':hover'
    });
    expect(testButton).toHaveStyleRule('--bg', '#000', {
      modifier: ':hover'
    });
  });

  it('renders a dark button', () => {
    //* Act
    const { getByTestId } = render(
      <Button dark data-testid='testButton'>
        Test Button
      </Button>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff');
    expect(testButton).toHaveStyleRule('--bg', '#000');
    // Hover test
    expect(testButton).toHaveStyleRule('--bg', '#333', {
      modifier: ':hover'
    });
  });

  it('renders a brand color button', () => {
    //* Act
    const { getByTestId } = render(
      <Button brand data-testid='testButton'>
        Test Button
      </Button>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff');
    expect(testButton).toHaveStyleRule('--bg', '#000');
    // Hover test
    expect(testButton).toHaveStyleRule('--bg', red, {
      modifier: ':hover'
    });
  });

  it('renders a transparent button', () => {
    //* Act
    const { getByTestId } = render(
      <Button trans data-testid='testButton'>
        Test Button
      </Button>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Transparency test
    expect(testButton).toHaveStyleRule('--txtColor', '#b3b3b3');
    expect(testButton).toHaveStyleRule('--bg', 'transparent');

    // Hover test
    expect(testButton).toHaveStyleRule('--txtColor', '#000', {
      modifier: ':hover'
    });
    expect(testButton).toHaveStyleRule('--bg', 'transparent', {
      modifier: ':hover'
    });
  });
});

describe('<ButtonLink /> tests', () => {
  it('renders a default button link', () => {
    //* Act
    const { getByTestId } = render(
      <Router>
        <ButtonLink to='/test' data-testid='testButton'>
          test link
        </ButtonLink>
      </Router>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Href test
    expect(testButton.getAttribute('href')).toBe('/test');

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#000');
    expect(testButton).toHaveStyleRule('color', 'var(--txtColor)');

    expect(testButton).toHaveStyleRule('--bg', '#fff');
    expect(testButton).toHaveStyleRule('background', 'var(--bg)');

    // Hover tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff', {
      modifier: ':hover'
    });
    expect(testButton).toHaveStyleRule('--bg', '#000', {
      modifier: ':hover'
    });
  });

  it('renders a dark button link', () => {
    //* Act
    const { getByTestId } = render(
      <Router>
        <ButtonLink dark to='/test' data-testid='testButton'>
          test link
        </ButtonLink>
      </Router>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff');
    expect(testButton).toHaveStyleRule('--bg', '#000');

    // Hover test
    expect(testButton).toHaveStyleRule('--bg', '#333', {
      modifier: ':hover'
    });
  });

  it('renders a brand color button link', () => {
    //* Act
    const { getByTestId } = render(
      <Router>
        <ButtonLink brand to='/test' data-testid='testButton'>
          Test Button
        </ButtonLink>
      </Router>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Color rules tests
    expect(testButton).toHaveStyleRule('--txtColor', '#fff');
    expect(testButton).toHaveStyleRule('--bg', '#000');
    // Hover test
    expect(testButton).toHaveStyleRule('--bg', red, {
      modifier: ':hover'
    });
  });

  it('renders a transparent button link', () => {
    //* Act
    const { getByTestId } = render(
      <Router>
        <ButtonLink trans to='/test' data-testid='testButton'>
          Test Button
        </ButtonLink>
      </Router>
    );

    //* Assert
    const testButton = getByTestId('testButton');

    // Snapshot test
    expect(testButton).toMatchSnapshot();

    // Transparency test
    expect(testButton).toHaveStyleRule('--txtColor', '#b3b3b3');
    expect(testButton).toHaveStyleRule('--bg', 'transparent');

    // Hover test
    expect(testButton).toHaveStyleRule('--txtColor', '#000', {
      modifier: ':hover'
    });
    expect(testButton).toHaveStyleRule('--bg', 'transparent', {
      modifier: ':hover'
    });
  });
});
