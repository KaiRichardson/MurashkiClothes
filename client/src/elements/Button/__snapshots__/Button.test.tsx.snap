// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> tests renders a brand color button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  background: var(--bg);
  color: var(--txtColor);
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #ee4266;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<Button /> tests renders a dark button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  background: var(--bg);
  color: var(--txtColor);
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #333;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<Button /> tests renders a default button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  background: var(--bg);
  color: var(--txtColor);
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<Button /> tests renders a transparent button 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  cursor: pointer;
  padding: 0.5rem 1rem;
  border: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  background: var(--bg);
  color: var(--txtColor);
  --txtColor: #b3b3b3;
  --bg: transparent;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --txtColor: #000;
  --bg: transparent;
}

<button
  class="c0"
  data-testid="testButton"
>
  Test Button
</button>
`;

exports[`<ButtonLink /> tests renders a brand color button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  text-align: center;
  display: inline-block;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #ee4266;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  Test Button
</a>
`;

exports[`<ButtonLink /> tests renders a dark button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  text-align: center;
  display: inline-block;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --bg: #333;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  test link
</a>
`;

exports[`<ButtonLink /> tests renders a default button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  text-align: center;
  display: inline-block;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  test link
</a>
`;

exports[`<ButtonLink /> tests renders a transparent button link 1`] = `
.c0 {
  --bg: #fff;
  --txtColor: #000;
  background: var(--bg);
  color: var(--txtColor);
  text-align: center;
  display: inline-block;
  border: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 250ms ease-in;
  transition: all 250ms ease-in;
  cursor: pointer;
  padding: 0.5rem 1rem;
  --txtColor: #b3b3b3;
  --bg: transparent;
}

.c0:hover {
  --bg: #000;
  --txtColor: #fff;
}

.c0:hover {
  --txtColor: #000;
  --bg: transparent;
}

<a
  class="c0"
  data-testid="testButton"
  href="/test"
>
  Test Button
</a>
`;
